### ISSUE TITLE
Getting a 504 with site query

### ISSUE THREAD

**Issue by jread-usgs** [NWQMC/WQP-WQX-Services#274]
Not sure if someone pointed this out already: https://github.com/USGS-R/dataRetrieval/issues/507

Seems https://www.waterqualitydata.us/data/Station/search?siteType=Lake%2C%20Reservoir%2C%20Impoundment&characteristicName=Temperature%2C%20water&zip=yes&mimeType=tsv times out with a cloudfront error

![image](https://user-images.githubusercontent.com/2349007/75406184-6d1b3780-58d5-11ea-96f3-fb0594877080.png)

**Comment by limnoliver** [comment]
I'm also having this same issue. Per Laura's suggestion, I tried looping through states to make a nation-wide `whatWQPdata` call, but it's getting hung up on data-rich states like Florida, e.g., for the call to:

https://www.waterqualitydata.us/data/Station/search?characteristicName=Temperature%3BTemperature%2C%20sample%3BTemperature%2C%20water%3BTemperature%2C%20water%2C%20deg%20F&statecode=US%3A12&zip=yes&mimeType=geojson

**Comment by jread-usgs** [comment]
Bummer, I can't get past even a low-ish data state (AR) with Salinity:

https://www.waterqualitydata.us/data/Station/search?characteristicName=Salinity&StateName=AR&zip=yes&mimeType=geojson

**Comment by mbucknell** [comment]
We don't support a StateName parameter. You can get the data with https://www.waterqualitydata.us/data/Station/search?statecode=US%3A04&characteristicName=Salinity&mimeType=geojson&zip=yes. 

Swagger docs for data retrieval can be found here: https://www.waterqualitydata.us/data/swagger-ui/index.html?url=/data/v3/api-docs#/Station/stationGetRequest

**Comment by jread-usgs** [comment]
Good catch, thanks Mary. Do you think in the future WQP will support queries w/o a geographic filter (like county or bbox)? We're unable to do a site query for a parameter w/o having to loop through some geo units after the switch to the cloud.

**Comment by mbucknell** [comment]
I believe both of those work now. I just tried it for a county and a bbox with geojson mimetype and got stations.

**Comment by limnoliver** [comment]
I think the problem with bbox is timing out if the bbox is too big (which is the same problem with using states).

**Comment by jkreft-usgs** [comment]
It is a challenge.  We are starting to reach the limits of our existing data structure as we climb past 400 million rows of result data, especially with the 60 second timeout. We have some ideas as to how to make things work better, but we are currently focused on controlling hosting costs so that there will be funding left to do these kinds of optimizations and improvements.

**Comment by jread-usgs** [comment]
Understood. Thanks Jim.
